expression/integration_test.go

```diff
@@ -3387,18 +3387,18 @@ func (s *testIntegrationSuite) TestCompareBuiltin(c *C) {
 	// for greatest
 	result = tk.MustQuery(`select greatest(1, 2, 3), greatest("a", "b", "c"), greatest(1.1, 1.2, 1.3), greatest("123a", 1, 2)`)
 	result.Check(testkit.Rows("3 c 1.3 123"))
-	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1265|Data Truncated"))
+	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1292|Truncated incorrect FLOAT value: '123a'"))
 	result = tk.MustQuery(`select greatest(cast("2017-01-01" as datetime), "123", "234", cast("2018-01-01" as date)), greatest(cast("2017-01-01" as date), "123", null)`)
 	// todo: MySQL returns "2018-01-01 <nil>"
 	result.Check(testkit.Rows("2018-01-01 00:00:00 <nil>"))
-	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1292|invalid time format: '123'", "Warning|1292|invalid time format: '234'", "Warning|1292|invalid time format: '123'"))
+	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|8036|invalid time format: '123'", "Warning|8036|invalid time format: '234'", "Warning|8036|invalid time format: '123'"))
 	// for least
 	result = tk.MustQuery(`select least(1, 2, 3), least("a", "b", "c"), least(1.1, 1.2, 1.3), least("123a", 1, 2)`)
 	result.Check(testkit.Rows("1 a 1.1 1"))
-	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1265|Data Truncated"))
+	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1292|Truncated incorrect FLOAT value: '123a'"))
 	result = tk.MustQuery(`select least(cast("2017-01-01" as datetime), "123", "234", cast("2018-01-01" as date)), least(cast("2017-01-01" as date), "123", null)`)
 	result.Check(testkit.Rows("123 <nil>"))
-	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1292|invalid time format: '123'", "Warning|1292|invalid time format: '234'", "Warning|1292|invalid time format: '123'"))
+	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|8036|invalid time format: '123'", "Warning|8036|invalid time format: '234'", "Warning|8036|invalid time format: '123'"))
```

Comment:
Why change this error code? MySQL error code is 1292.
```
mysql> select least(cast("2017-01-01" as datetime), "123", "234", cast("2018-01-01" as date)), least(cast("2017-01-01" as date), "123", null);
+---------------------------------------------------------------------------------+------------------------------------------------+
| least(cast("2017-01-01" as datetime), "123", "234", cast("2018-01-01" as date)) | least(cast("2017-01-01" as date), "123", null) |
+---------------------------------------------------------------------------------+------------------------------------------------+
| 123                                                                             | NULL                                           |
+---------------------------------------------------------------------------------+------------------------------------------------+
1 row in set, 3 warnings (0.02 sec)

mysql> show warnings;
+---------+------+--------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                              |
+---------+------+--------------------------------------------------------------------------------------+
| Warning | 1292 | Incorrect datetime value: '123' for column 'cast("2017-01-01" as datetime)' at row 1 |
| Warning | 1292 | Incorrect datetime value: '234' for column 'cast("2017-01-01" as datetime)' at row 1 |
| Warning | 1292 | Incorrect date value: '123' for column 'cast("2017-01-01" as date)' at row 1         |
+---------+------+--------------------------------------------------------------------------------------+
3 rows in set (0.01 sec)
```

Reply:
Yes, but the error message is really different from MySQL. I think we could combine these errors together.